src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_gligen.py

```diff
@@ -0,0 +1,827 @@
+            do_classifier_free_guidance,
+            negative_prompt,
+            prompt_embeds=prompt_embeds,
+            negative_prompt_embeds=negative_prompt_embeds,
+        )
+
+        # 4. Prepare timesteps
+        self.scheduler.set_timesteps(num_inference_steps, device=device)
+        timesteps = self.scheduler.timesteps
+
+        # 5. Prepare latent variables
+        num_channels_latents = self.unet.in_channels
+        latents = self.prepare_latents(
+            batch_size * num_images_per_prompt,
+            num_channels_latents,
+            height,
+            width,
+            prompt_embeds.dtype,
+            device,
+            generator,
+            latents,
+        )
+
+        # 5.1 Prepare GLIGEN variables
+        if gligen_phrases is not None:
+            assert len(gligen_phrases) == len(gligen_boxes)
+            max_objs = 30
+            if len(gligen_boxes) > max_objs:
+                warnings.warn(
+                    f"More that {max_objs} objects found. Only first {max_objs} objects will be processed.",
+                    FutureWarning,
+                )
+                gligen_phrases = gligen_phrases[:max_objs]
+                gligen_boxes = gligen_boxes[:max_objs]
+            _boxes = gligen_boxes
+            # prepare batched input to the PositionNet (boxes, phrases, mask)
+            # Get tokens for phrases from pre-trained CLIPTokenizer
+            tokenizer_inputs = self.tokenizer(gligen_phrases, padding=True, return_tensors="pt").to(device)
+            # For the token, we use the same pre-trained text encoder
+            # to obtain its text feature
+            _text_embeddings = self.text_encoder(**tokenizer_inputs).pooler_output
+            n_objs = min(len(_boxes), max_objs)
+            dtype = self.text_encoder.dtype
+            # For each entity, described in phrases, is denoted with a bounding box,
+            # we represent the location information as (xmin,ymin,xmax,ymax)
+            boxes = torch.zeros(max_objs, 4, device=device, dtype=dtype)
+            boxes[:n_objs] = torch.tensor(_boxes[:n_objs])
+            text_embeddings = torch.zeros(max_objs, self.unet.cross_attention_dim, device=device, dtype=dtype)
+            text_embeddings[:n_objs] = _text_embeddings[:n_objs]
+            # Generate a mask for each object that is entity described by phrases
+            masks = torch.zeros(max_objs, device=device, dtype=dtype)
+            masks[:n_objs] = 1
+
+            repeat_batch = batch_size * num_images_per_prompt
+            if do_classifier_free_guidance:
+                repeat_batch = repeat_batch * 2
+            boxes = boxes.unsqueeze(0).expand(repeat_batch, -1, -1).clone()
+            text_embeddings = text_embeddings.unsqueeze(0).expand(repeat_batch, -1, -1).clone()
+            masks = masks.unsqueeze(0).expand(repeat_batch, -1).clone()
+            if do_classifier_free_guidance:
+                masks[: repeat_batch // 2] = 0
```

Comment:
Fixed it by using your recommendation, but I retained `repeat_batch` because it's used down below in the code.

using `batch_size * num_images_per_prompt` down below wouldn't be right in `do_classifier_free_guidance` case